### PR TITLE
nu-cli/completions: add completion for $env.

### PR DESCRIPTION
**Pull request by herlon214**
# Description

This is a simple implementation for adding environment variables completions when using `$env.`.

We can implement the completion for the other variables if the solution for this one is correct.

[asciicast](https://asciinema.org/a/jwj38qvGoYotoFMH1IVHoZPfB)

# Tests

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

**Comment by rgwood** [comment]
This is very useful, thank you!

**Review by kubouch — CHANGES_REQUESTED** [review]

File: crates/nu-protocol/src/engine/engine_state.rs [hunk 1]
```diff
@@ -1233,6 +1233,16 @@ impl<'a> StateWorkingSet<'a> {
         self.permanent_state.env_vars.get(name)
     }
 
+    pub fn list_env(&self) -> Vec<String> {
+        let mut env_vars = vec![];
+
+        for env_var in self.permanent_state.env_vars.clone().into_iter() {
```
[the remark this answers is not in the document]
Comment (herlon214) [reply crates/nu-protocol/src/engine/engine_state.rs:1239]: How can I access it? I think the working set and the engine state doesn't have the stack field

[the remark this answers is not in the document]
Comment (herlon214) [reply crates/nu-protocol/src/engine/engine_state.rs:1239]: Perhaps:
```
            let stack = Stack::default();

            for env_var in stack.get_env_vars(&self.engine_state) {
                output.push(Suggestion {
                    value: env_var.0,
                    description: None,
                    extra: None,
                    span: reedline::Span {
                        start: span.start - offset,
                        end: span.end - offset,
                    },
                });
            }
```

I'm not sure if creating a new stack is the correct way to do it

[the remark this answers is not in the document]
Comment (kubouch) [reply crates/nu-protocol/src/engine/engine_state.rs:1239]: You'd need to use the runtime stack that is being passed around during evaluation. See the NuCompleter struct, it gets the stack when you create it, so I'd guess you need to pass it to the VaribleCompletion.

[the remark this answers is not in the document]
Comment (kubouch) [reply crates/nu-protocol/src/engine/engine_state.rs:1239]: The default stack won't work.

[the remark this answers is not in the document]
Comment (herlon214) [reply crates/nu-protocol/src/engine/engine_state.rs:1239]: Got it, I updated the code.

**Comment by herlon214** [comment]
@kubouch giving a `Vec<u8>` for the previous expr variable, can you give me some tip about how to find the variable in the engine state? 
I saw the `get_var` uses the id but at this point I don't have the ID for it

File: crates/nu-protocol/src/engine/engine_state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kubouch) [reply crates/nu-protocol/src/engine/engine_state.rs:1239]: @herlon214 OK, cool, your latest commit should work. You can try by `let FOO = "foo"`, then `hide FOO`, then `$env.<tab>` and there shouldn't be `FOO` in the completion list.

**Comment by kubouch** [comment]
@herlon214 You need to use `engine_state.find_variable(name: &[u8])` which gives you (maybe) the VarId.

**Comment by kubouch** [comment]
There is no `find_variable()` in EngineState but you can make one.

**Comment by herlon214** [comment]
@kubouch nice, it worked!

I'm trying to also make completions for `$any.` vars, but when I read the `var.ty` it shows `Any` for `let test = {"a": 1, "b": 2}
`, shouldn't it be a `List`?

**Comment by herlon214** [comment]
Also, shouldn't the List be a Vec?
```
let something = ["a","b","c"]
```
In the Type it's actually a `List(Box<Type>)`, how can I get the values for it?

**Comment by kubouch** [comment]
@herlon214 I don't know exactly what you're asking (I'm not very familiar with the completions code). I'd suggest landing this PR as only `$env` completions. Then, you can open a new one for variables where we can see only the changes related to variable completions.

**Review by kubouch — APPROVED** [review]

**Comment by kubouch** [comment]
@herlon214 I tested your env completions and they seem to work fine, even with the hiding. So if you're OK with it, I would land this and let's bring the variables discussion to Discord or a new PR.

**Comment by herlon214** [comment]
@kubouch that sounds good

**Comment by kubouch** [comment]
OK, thanks a lot! This is quite awesome, I always wanted to have (environment) variable completions.

**Pull request merged**